### DISCUSSION TITLE
Feature: Expandable rows / "Item details"

### DISCUSSION THREAD

**Discussion by AtishaGarinMchd** [revolist/revogrid#648] [General]
It would be great to have other fields visible after an item click.

[One example here](https://vaadin.com/docs/latest/components/grid/#item-details).

[This example](https://www.ag-grid.com/javascript-data-grid/master-detail-height/) is one level of complexity further (set of rows) and such feature could be separate, but there is still the concept of expandable "additional details" for a given row.

**Comment by atsepkov** [comment]
You can already accomplish this by hooking into the revogrid events (https://revolist.github.io/revogrid/guide/api.html#properties). In particular, `beforecellfocus` can be used to render the selected cell/row differently from the other ones. I use this technique myself to hide/unhide additional rows when user clicks on a cell. Here is an example of this in action: https://investomation.com/tools/napkin (row names with "+" in them can be expanded to show more rows).

**Comment by revolist** [comment]
We have a lot of ways to do it, good starting point is a tree view, you can trigger event and update dataset.
Recommended way to do it would be to use pro plugin we created for this matter: Tree data view https://rv-grid.com/pro/
